File: backend/Crystal.Core/Entities/ScheduledShift.cs
namespace Crystal.Core.Entities;

public class ScheduledShift
{
    public int Id { get; set; }

    public int? EmployeeProfileId { get; set; }
    public EmployeeProfile? EmployeeProfile { get; set; }

    public int? LocationId { get; set; }
    public Location? Location { get; set; }

    public int JobPositionId { get; set; }
    public JobPosition JobPosition { get; set; } = null!;

    public DateOnly Date { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }

    public bool IsDeleted { get; set; } = false;

    public ICollection<TimeEntry> TimeEntries { get; set; } = new List<TimeEntry>();
}

File: frontend/src/data/types/hr/scheduledShift.ts
export interface ScheduledShift {
    id: number;
    employeeProfileId?: number | null;
    employeeFirstName?: string | null;
    employeeLastName?: string | null;
    jobPositionId?: number | null;
    jobPositionName?: string | null;
    jobPositionColor?: string | null;
    locationId?: number | null;
    locationTitle?: string | null;
    date: string;
    startTime: string;
    endTime: string;
    isDeleted: boolean;
}

export interface ScheduledShiftApiDto {
    id: number;
    employeeProfileId?: number | null;
    employeeFirstName?: string | null;
    employeeLastName?: string | null;
    jobPositionId?: number | null;
    jobPositionName?: string | null;
    jobPositionColor?: string | null;
    locationId?: number | null;
    locationTitle?: string | null;
    date: string;
    startTime: string;
    endTime: string;
}

export interface ScheduledShiftFormData {
    employeeProfileId?: number | null;
    jobPositionId?: number | null;
    locationId?: number;
    date: string;
    startTime: string;
    endTime: string;
}
